cloud-content/impl/src/main/java/coza/opencollab/sakai/cloudcontent/SwiftFileSystemHandler.java

```diff
@@ -0,0 +1,438 @@
+package coza.opencollab.sakai.cloudcontent;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Closeables;
+import com.google.common.io.CountingInputStream;
+import com.google.inject.Module;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.jclouds.apis.ApiMetadata;
+import org.jclouds.ContextBuilder;
+import org.jclouds.http.options.GetOptions;
+import org.jclouds.io.Payload;
+import org.jclouds.io.Payloads;
+import org.jclouds.logging.slf4j.config.SLF4JLoggingModule;
+import org.jclouds.openstack.swift.v1.SwiftApi;
+import org.jclouds.openstack.swift.v1.SwiftApiMetadata;
+import org.jclouds.openstack.swift.v1.domain.SwiftObject;
+import org.jclouds.openstack.swift.v1.features.ContainerApi;
+import org.jclouds.openstack.swift.v1.features.ObjectApi;
+import org.jclouds.openstack.swift.v1.options.CreateContainerOptions;
+import org.jclouds.openstack.swift.v1.options.PutOptions;
+import org.jclouds.osgi.ApiRegistry;
+import org.sakaiproject.content.api.FileSystemHandler;
+import org.springframework.util.FileCopyUtils;
+
+/**
+ * The cloud implementation of FileSystemHandler.
+ * <p/>
+ * This class read and write files to and from OpenStack-Swift cloud storage.
+ *
+ * @author OpenCollab
+ */
+public class SwiftFileSystemHandler implements FileSystemHandler {
+
+    /**
+     * The provider for the jcloud storage.
+     */
+    private static final String CLOUD_PROVIDER = "openstack-swift";
+
+    /**
+     * The ApiMetadata class to register and use for Swift.
+     */
+    private static final ApiMetadata CLOUD_API_METADATA = new SwiftApiMetadata();
+    /**
+     * The connection endpoint to the swift storage.
+     */
+    private String endpoint;
+    /**
+     * The identity of the user to connect to swift.
+     */
+    private String identity;
+    /**
+     * The credential for the user.
+     */
+    private String credential;
+    /**
+     * The region to connect to.
+     */
+    private String region;
+    /**
+     * The swift storage api.
+     */
+    private SwiftApi swiftApi;
+    /**
+     * The base container to store everything under.
+     * If this is not set then the filePath will specify the base container.
+     */
+    private String baseContainer;
+    /**
+     * Whether to delete empty containers after a resource delete.
+     */
+    private boolean deleteEmptyContainers = false;
+    /**
+     * Whether to use the id for the file path.
+     */
+    private boolean useIdForPath = false;
+    /**
+     * The regular expression for all the characters that is not valid 
+     * for container and resource names.
+     */
+    private String invalidCharactersRegex = "[:*?<|>]";
+    /**
+     * The logger for warnings and errors.
+     */
+    private Logger logger = new DefaultLogger();
```

Comment:
Don't we want a default logger that does a little bit of logging instead of throwing everything away?